client/source/js/modules/ui/type-selector/type-selector-ctrl.js

```diff
@@ -15,11 +16,32 @@ define(['./module', 'underscore'], function (module, _) {
       }
 
       $scope.$on('$stateChangeSuccess',function() {
-        showPartInViews('typeSelector', [
```

Comment:
Rather than creating another method I described it in the controller comment as what happens here is the core of the controllers purpose.

Does that sound good as well?